Core/src/org/sleuthkit/autopsy/centralrepository/eventlisteners/CaseEventListener.java

```diff
@@ -154,82 +206,90 @@ private ContentTagTask(CentralRepository db, PropertyChangeEvent evt) {
 
         @Override
         public void run() {
-            if (!CentralRepository.isEnabled()) {
+            if (!CentralRepository.isEnabled())
                 return;
-            }
 
-            AbstractFile af;
-            TskData.FileKnown knownStatus;
-            String comment;
             if (Case.Events.valueOf(event.getPropertyName()) == Case.Events.CONTENT_TAG_ADDED) {
-                // For added tags, we want to change the known status to BAD if the 
-                // tag that was just added is in the list of central repo tags.
-                final ContentTagAddedEvent tagAddedEvent = (ContentTagAddedEvent) event;
-                final ContentTag tagAdded = tagAddedEvent.getAddedTag();
-
-                if (TagsManager.getNotableTagDisplayNames().contains(tagAdded.getName().getDisplayName())) {
-                    if (tagAdded.getContent() instanceof AbstractFile) {
-                        af = (AbstractFile) tagAdded.getContent();
-                        knownStatus = TskData.FileKnown.BAD;
-                        comment = tagAdded.getComment();
-                    } else {
-                        LOGGER.log(Level.WARNING, "Error updating non-file object");
-                        return;
-                    }
-                } else {
-                    // The added tag isn't flagged as bad in central repo, so do nothing
-                    return;
-                }
-            } else { // CONTENT_TAG_DELETED
-                // For deleted tags, we want to set the file status to UNKNOWN if:
-                //   - The tag that was just removed is notable in central repo
-                //   - There are no remaining tags that are notable 
-                final ContentTagDeletedEvent tagDeletedEvent = (ContentTagDeletedEvent) event;
+                handleTagAdded((ContentTagAddedEvent) event);
+            } 
+            else { 
+                handleTagDeleted((ContentTagDeletedEvent) event);
+            }
+        }
+        
+        private void handleTagDeleted(ContentTagDeletedEvent tagDeletedEvent) {
+            // CONTENT_TAG_DELETED
+            // For deleted tags, we want to set the file status to UNKNOWN if:
+            //   - The tag that was just removed is notable in central repo
+            //   - There are no remaining tags that are notable 
+            //   - If there are other notable tags, the comment should be updated to most recent tag
```

Comment:
This is tricky code. Again, I think we want to be very careful in our comments...

This first comment describes what the method does. It belongs in a header comment.

The logic in the method does two things:

1) We are not setting the status of the file that has been tagged. We are setting the known status of the correlation attribute instance in the CR corresponding to the tagged file. We want to change that status from notable to unknown if the tag that was deleted was the only tag on the file that confers notability. 
2) If the deleted tag conferred notability on the file and also had a comment, we want to remove the comment from the correlation attribute instance, if and only if the comment on the correlation attribute instance and the tag match. Otherwise, we do not know the source of the comment (it could have been applied through the add/edit CR comment action independent of any tagging) and will not remove the comment.